libarchive/archive_write_disk_posix.c

```diff
@@ -4044,10 +4044,10 @@ older(struct stat *st, struct archive_entry *entry)
 {
 	/* First, test the seconds and return if we have a definite answer. */
 	/* Definitely older. */
-	if (st->st_mtime < archive_entry_mtime(entry))
+	if ((time_t) st->st_mtime < archive_entry_mtime(entry))
```

Comment:
There's also a style error here:  Please don't put a space between a cast and the value.